### PR TITLE
Use macOS-specific application icon

### PR DESCRIPTION
**Pull request by claucambra**
Signed-off-by: Claudio Cambra <claudio.cambra@gmail.com>

### CODE REVIEW

**Comment by elsiehupp** [comment]
Is this for the purpose of facilitating #4367? (As I read it this pull request is basically just clarifying that the particular SVG in question is only used for the Finder sidebar and not, say, the Nautilus sidebar, which wasn't terribly clear from the previous nomenclature.)

**Comment by claucambra** [comment]
> Is this for the purpose of facilitating #4367? (As I read it this pull request is basically just clarifying that the particular SVG in question is only used for the Finder sidebar and not, say, the Nautilus sidebar, which wasn't terribly clear from the previous nomenclature.)

No, this is to set the application's icon to your Big Sur-style icon in #4631

**Comment by elsiehupp** [comment]
On some level I think it would be useful to clarify in the filename that the sidebar icon is specifically used by the Finder, but it could also make sense make the nomenclature more generic, like `Nextcloud-{platform}-shell-sidebar`, if you wanted to facilitate slight variations for, e.g., Windows Explorer, Nautilus, Dolphin, elementary Files, etc. (How the distinction for different Linux shells would be implemented would probably be a bit more complicated than, say, the distinction between Windows and macOS.)

**Comment by elsiehupp** [comment]
> > Is this for the purpose of facilitating #4367? (As I read it this pull request is basically just clarifying that the particular SVG in question is only used for the Finder sidebar and not, say, the Nautilus sidebar, which wasn't terribly clear from the previous nomenclature.)
> 
> No, this is to set the application's icon to your Big Sur-style icon in #4631

Ah, okay!

**Review by elsiehupp — APPROVED** [review]

**Review by elsiehupp — APPROVED** [review]

**Comment by claucambra** [comment]
> In the case of Linux shells that use [the XDG Icon Theme Specification](https://specifications.freedesktop.org/icon-theme-spec/), it could work to assign the Nextcloud folder a filesystem attribute with the icon name and then provide the Nextcloud (or vendor) logo as icon, though I'm not 100% sure how that would work with sandboxed versions of Nextcloud Desktop such as Flatpak.
> 
> This is a bit of a digression, but again my main point here is that `Nextcloud-{platform}-shell-sidebar` could potentially facilitate better shell-sidebar icon support on platforms other than macOS.

I think this naming structure would make things clearer, but it would also require changes to our branding infra (which I am not super familiar with). Worth discussing though :)

File: NEXTCLOUD.cmake [hunk 1]
```diff
@@ -5,7 +5,14 @@ set( APPLICATION_DOMAIN     "nextcloud.com" )
 set( APPLICATION_VENDOR     "Nextcloud GmbH" )
 set( APPLICATION_UPDATE_URL "https://updates.nextcloud.org/client/" CACHE STRING "URL for updater" )
 set( APPLICATION_HELP_URL   "" CACHE STRING "URL for the help menu" )
-set( APPLICATION_ICON_NAME  "Nextcloud" )
+
+if(APPLE AND APPLICATION_NAME STREQUAL "Nextcloud")
+    set( APPLICATION_ICON_NAME "Nextcloud-macOS" )
+    message("Using macOS-specific application icon: ${APPLICATION_ICON_NAME}")
+else()
```
Comment (mgallien) [inline NEXTCLOUD.cmake:12]: but should we not also offer this better macOS icon feature when `APPLICATION_NAME` is not `Nextcloud` ?

Comment (mgallien) [reply NEXTCLOUD.cmake:12]: for the other question related to the check on `APPLICATION_NAME` @claudio would it be possible to check for the existence of the icon file ?

Comment (claucambra) [reply NEXTCLOUD.cmake]: Yeah, added a check for the existence of the icon name

**Review by mgallien — APPROVED** [review]

File: NEXTCLOUD.cmake (hunk 1 above)
Comment (elsiehupp) [reply NEXTCLOUD.cmake]: You can find the updated version of the icon at #4714.

**Pull request merged**